db/version_set.h

```diff
@@ -1476,8 +1489,9 @@ class VersionSet {
   // The returned WalSet needs to be accessed with DB mutex held.
   const WalSet& GetWalSet() const { return wals_; }
 
-  void NewManifestOnNextUpdate() {
-    new_manifest_on_next_update_.store(true, std::memory_order_relaxed);
+  // Called when replication epoch has changed
+  void UpdateReplicationEpoch(uint64_t new_replication_epoch) {
+    next_replication_epoch_ = new_replication_epoch;
```

Comment:
Nit: use .emplace() to make it clear to the code reader that `next_replication_epoch_` is std::optional.